interfaces/acados_template/acados_template/acados_ocp.py

```diff
@@ -1303,18 +1311,44 @@ def translate_to_feasibility_problem(self, keep_x0=False, keep_cost=False) -> No
         model.con_r_expr_e = None
         model.con_r_in_phi_e = None
 
+        # Convert initial conditions to l2 penalty
+        if parametric_x0:
+            symbol = self.get_casadi_symbol(self.model.x)
+            param_lbx0 = symbol('x0', len(constraints.idxbx_0))
+            if keep_x0:
+                param_ubx0 = []
+            else:
+                param_ubx0 = symbol('x0', len(constraints.idxbx_0))
+            model.p = ca.vertcat(model.p, param_lbx0, param_ubx0)
+
+            # Set parameters
+            if keep_x0:
+                self.parameter_values = constraints.lbx_0
+            else:
+                self.parameter_values = np.concatenate((constraints.lbx_0, constraints.ubx_0))
+
+
         expr_bound_list_0 = [
             (model.u[constraints.idxbu], constraints.lbu, constraints.ubu),
             (model.con_h_expr_0, constraints.lh_0, constraints.uh_0),
         ]
 
+
+
         if keep_x0:
             if constraints.has_x0:
-                new_constraints.x0 = constraints.lbx_0
+                if parametric_x0 and constraints.has_x0:
```

Comment:
I would keep the option that both can be true. In case, we would like to solve the feasibility problem with an SQP solver, then it makes sense to keep the constraint as hard constraint, but still have the possibility to make it parametric.